Mathlib/RingTheory/HopfAlgebra/GroupLike.lean

```diff
@@ -0,0 +1,66 @@
+/-
+Copyright (c) 2025 Yaël Dillies, Michał Mrugała. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Michał Mrugała
+-/
+import Mathlib.RingTheory.HopfAlgebra.Basic
+import Mathlib.RingTheory.Bialgebra.GroupLike
+
+/-!
+# Group-like elements in a Hopf algebra
+
+This file proves that group-like elements in a Hopf algebra form a group.
+-/
+
+open HopfAlgebra
+
+variable {R A : Type*}
+
+section Semiring
+variable [CommSemiring R] [Semiring A] [HopfAlgebra R A] {a b : A}
+
+@[simp] lemma IsGroupLikeElem.antipode_mul_cancel (ha : IsGroupLikeElem R a) :
+    antipode R a * a = 1 := by
+  simpa [ha, -mul_antipode_lTensor_comul_apply] using mul_antipode_rTensor_comul_apply (R := R) a
+
+@[simp] lemma IsGroupLikeElem.mul_antipode_cancel (ha : IsGroupLikeElem R a) :
+    a * antipode R a = 1 := by
+  simpa [ha, -mul_antipode_lTensor_comul_apply] using mul_antipode_lTensor_comul_apply (R := R) a
+
+variable (R) in
+/-- Turn a group-like element `a` into a unit with inverse its antipode. -/
+@[simps]
+def Units.ofIsGroupLikeElem (a : A) (ha : IsGroupLikeElem R a) : Aˣ where
+  val := a
+  inv := antipode R a
+  val_inv := ha.mul_antipode_cancel
+  inv_val := ha.antipode_mul_cancel
+
+lemma IsGroupLikeElem.isUnit (ha : IsGroupLikeElem R a) : IsUnit a :=
```

Comment:
Is this true in some weaker setting (eg for Bialgebra) as `IsRegular`?

Reply:
No. Consider eg `R[M]` for `M` a monoid. For `m : M`, `of m` is a group-like element (I have formalised this, and it will be in a later PR) but is a unit iff `m` itself is a unit

Reply:
Sorry, I misread your message. I will change my answer to "I have no idea how to prove/disprove this"

Reply:
Doesn't my counterexample of `R[M]` with `M` a non-cancellative monoid work?

Reply:
Yes, I said that because you haven't formalised that and also haven't resolved the comment, so my comment above meant "let's resolve this conversation, we don't have to prove it here"